codegen-core/src/main/kotlin/software/amazon/smithy/rust/codegen/core/smithy/generators/EnumGenerator.kt

```diff
@@ -198,14 +206,28 @@ open class EnumGenerator(
         }
     }
 
+    private fun unknownVariantValue(): RuntimeType {
+        return RuntimeType.forInlineFun(UnknownVariantValue, RustModule.Model) {
+            rust("##[allow(missing_docs)]")
```

Comment:
Good point. I tried removing `pub` from `pub struct UnknownVariantValue(String);` (so that we don't have to add `#[allow(missing_docs)]`) but got
```
error[E0446]: private type `UnknownVariantValue` in public interface
    |
    |     Unknown(crate::model::UnknownVariantValue),
    |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ can't leak private type
```
It is therefore marked as `pub`. I wonder if we should add some docs instead of `#[allow(missing_docs)]`?